core/src/main/scala/kafka/coordinator/group/MemberMetadata.scala

```diff
@@ -66,7 +66,7 @@ private[group] class MemberMetadata(var memberId: String,
   var assignment: Array[Byte] = Array.empty[Byte]
   var awaitingJoinCallback: JoinGroupResult => Unit = null
   var awaitingSyncCallback: SyncGroupResult => Unit = null
-  var latestHeartbeat: Long = -1
+  var heartbeatSatisfied: Boolean = false
```

Comment:
May be useful to add a comment since this has a slightly unusual usage

Reply:
I will add a comment. I struggled to find a simpler way to map heartbeat expiration onto purgatory semantics. This was the best I could come up with, though I agree it is a little unusual.

Reply:
I had a different idea about this, but after second thought I think it's better to just make it hard-explicit, rather than embedding in a less-readable math equation. Now I feel it's better to be unusual than bug-vulnerable.